apps/oc_chef_wm/src/oc_chef_wm_named_policy.erl

```diff
@@ -0,0 +1,205 @@
+                invalid  ->
+                    oc_chef_wm_policies:policy_name_invalid(Req, State)
+            end;
+        _DeleteOrGet ->
+            validate_policy(false, OrgId, PolicyName, DbContext, Req, State, PolicyState)
+    end.
+validate_policy(CanCreate, OrgId, PolicyName, DbContext, Req,
+               #base_state{} = State, PolicyState = #policy_state{ oc_chef_policy = InputPolicy }) ->
+    case {chef_db:fetch(InputPolicy, DbContext), CanCreate} of
+		{not_found, true} ->
+            {{create_in_container, policies}, Req, State#base_state{resource_state = PolicyState#policy_state{create_policy = true}}};
+		{not_found, false} ->
+			Message = chef_wm_util:error_message_envelope(iolist_to_binary(["Cannot load policy ",
+																			PolicyName])),
+			Req1 = chef_wm_util:set_json_body(Req, Message),
+			{{halt, 404}, Req1, State#base_state{log_msg = policy_not_found}};
+		{forbidden, _} ->
+            Message = chef_wm_util:error_message_envelope(iolist_to_binary(["No permission for policy ",
+                                                                            PolicyName])),
+            Req1 = chef_wm_util:set_json_body(Req, Message),
+            {{halt, 403}, Req1, State#base_state{log_msg = policy_not_found}};
+		{#oc_chef_policy{authz_id = AuthzId} = Policy, _} ->
+            PolicyState1 = PolicyState#policy_state{oc_chef_policy = Policy},
+            State1 = State#base_state{resource_state = PolicyState1},
+            {{object, AuthzId}, Req, State1}
+    end.
+
+resource_exists(Req, State) ->
+    {true, Req, State}.
+
+to_json(Req, #base_state{
+                organization_name = OrgName,
+                resource_state = #policy_state{
+                                             oc_chef_policy = Policy
+                                            }} = State) ->
+    EncodedEjson = oc_chef_policy:assemble_policy_ejson(Policy, OrgName),
+
+    {EncodedEjson, Req, State}.
+
+from_json(Req, #base_state{resource_state = #policy_state{
+                                               create_policy = false,
+                                               oc_chef_policy = Policy,
+                                               policy_data = PolicyData
+                                              }
+                          } = State) ->
+    oc_chef_wm_base:update_from_json(Req, State, Policy , PolicyData);
+from_json(Req, #base_state{resource_state = #policy_state{
+                                               create_policy = true,
+                                               policy_authz_id = AuthzId,
+                                               policy_data = PolicyData
+                                              }
+                          } = State) ->
+    oc_chef_wm_base:create_from_json(Req, State, oc_chef_policy, {authz_id, AuthzId}, PolicyData).
+
+conflict_message(_Name) ->
+    {[{<<"error">>, <<"Policy already exists">>}]}.
```

Comment:
Since we have the name, does it make sense to include it in the message? "The policy '$Name' already exists"